### PR TITLE
fix for cpp api Transaction::redirectTo

### PR DESCRIPTION
**Pull request by dmorilha**

### CODE REVIEW

File: lib/cppapi/Transaction.cc [hunk 1]
```diff
@@ -397,9 +397,11 @@ Transaction::getCacheStatus()
 void
 Transaction::redirectTo(std::string const &url)
 {
-  std::string s = url;
   // Must re-alloc the string locally because ownership is transferred to the transaction.
-  TSHttpTxnRedirectUrlSet(state_->txn_, s.c_str(), url.length());
+  char * const buffer = reinterpret_cast<char *>(TSmalloc(url.size() + 1));
+  TSstrlcpy(buffer, url.c_str(), url.size());
```
[the remark this answers is not in the document]
Comment (dmorilha) [reply lib/cppapi/Transaction.cc]: most likely I just copied from somewhere, changing it

**Comment by bryancall** [comment]
[approve ci]
[add to whitelist]

**Comment by bryancall** [comment]
Cherry picked to 8.0.x

**Comment by zwoop** [comment]
@dmorilha Do we want this in 7.1.5? If so, can you please make a PR against the 7.1.x branch? The way to do this is to "git cherry-pick -x ..." and then resolve merge conflicts (if any), before pushing upstream. Then make sure to pick the 7.1.x branch on the PR of course.

**Comment by dmorilha** [comment]
I think we are good to skip it. thanks for checking

**Pull request merged**